src/troute-nwm/src/nwm_routing/preprocess.py

```diff
@@ -1094,10 +1096,12 @@ def nwm_forcing_preprocess(
         reservoir_usgs_df = (
             usgs_df_15min.join(link_lake_df, how = 'inner').
             reset_index().
-            set_index('usgs_lake_id').
-            drop(['index'], axis = 1)
+            set_index('usgs_lake_id')
         )
         
+        if 'index' in reservoir_usgs_df.columns:
+            reservoir_usgs_df=reservoir_usgs_df.drop(['index'], axis = 1)
+
```

Comment:
I think an alternative solution here would be to use the `drop=` parameter in `.reset_index()`. So:

```
        link_lake_df = (
            gage_lake_df.
            join(gage_link_df, how = 'inner').
            reset_index(drop=True).set_index('link')
        )
```

I'm not sure the cause of this issue (pandas version maybe?), but what was happening is that `reset_index()` makes the existing index a new column in the dataframe, sometimes naming it 'index', sometimes not. But either way, we want to remove that index all together, so using `reset_index(drop=True)` will do this regardless of the index name.

Reply:
@shorvath-noaa This sounds like a better, more comprehensive fix. 
@TrupeshKumarPatel, @JoshCu -- suggest we split these fixes into three PRs -- @shorvath-noaa, could you confirm that this would help?

Reply:
@jameshalgren @TrupeshKumarPatel [@JoshCU](https://github.com/JoshCu) yeah, I think splitting this up into separate PRs would help us test/approve these quickly.

Reply:
While we're modifying requirements.txt, should we add 
1. deprecated
2.  pyarrow
3.  geopandas?

They're currently being manually installed in a bunch of docker images and the [ngen documentation](https://github.com/NOAA-OWP/ngen/blob/master/doc/PYTHON_ROUTING.md#setup-virtual-environment) too.

Wheel is also needed for building the wheels but I don't know if it's strictly necessary to run t-troute. Might be a small boost to the install speed of some of the other required packages though.

Reply:
I've got some CI for building and testing a t-route docker image , but it needs the Lower/Colorado fix. Once it's merged I'll open a PR for the docker image

Reply:
I'd say yes to updating the requirements.txt. And I'm not too familiar with docker images, but we are working on getting a working version of LowerColorado_TX with nextgen, so I'll update you once that's ready.

Reply:
@shorvath-noaa, the only reason I put the `if` condition is because, as per the previous code, the first `index` dropping was applied on `link_lake_df` DataFrame and then re-applied `index` dropping after joining operation. Now, because the `index` column was already dropped, it couldn’t find the next join. Therefore, if you do `reset_index(drop=True)`, you will get the same error. 

So, I decided to add `if` conditions on both DataFrame to avoid error in any case.